riverdriver/riverpgxv5/river_pgx_v5_driver.go

```diff
@@ -272,13 +272,61 @@ func (e *Executor) JobInsertFull(ctx context.Context, params *riverdriver.JobIns
 		ScheduledAt: params.ScheduledAt,
 		State:       dbsqlc.RiverJobState(params.State),
 		Tags:        params.Tags,
+		UniqueKey:   params.UniqueKey,
 	})
 	if err != nil {
 		return nil, interpretError(err)
 	}
 	return jobRowFromInternal(job)
 }
 
+func (e *Executor) JobInsertUnique(ctx context.Context, params *riverdriver.JobInsertUniqueParams) (*riverdriver.JobInsertUniqueResult, error) {
+	insertRes, err := e.queries.JobInsertUnique(ctx, e.dbtx, &dbsqlc.JobInsertUniqueParams{
+		Args:        params.EncodedArgs,
+		CreatedAt:   params.CreatedAt,
+		Kind:        params.Kind,
+		MaxAttempts: int16(min(params.MaxAttempts, math.MaxInt16)),
+		Metadata:    params.Metadata,
+		Priority:    int16(min(params.Priority, math.MaxInt16)),
+		Queue:       params.Queue,
+		ScheduledAt: params.ScheduledAt,
+		State:       dbsqlc.RiverJobState(params.State),
+		Tags:        params.Tags,
+		UniqueKey:   params.UniqueKey,
+	})
+	if err != nil {
+		return nil, interpretError(err)
+	}
+
+	jobRow, err := jobRowFromInternal(&dbsqlc.RiverJob{
+		ID:          insertRes.ID,
+		Args:        insertRes.Args,
+		Attempt:     insertRes.Attempt,
+		AttemptedAt: insertRes.AttemptedAt,
+		AttemptedBy: insertRes.AttemptedBy,
+		CreatedAt:   insertRes.CreatedAt,
+		Errors:      insertRes.Errors,
+		FinalizedAt: insertRes.FinalizedAt,
+		Kind:        insertRes.Kind,
+		MaxAttempts: insertRes.MaxAttempts,
+		Metadata:    insertRes.Metadata,
+		Priority:    insertRes.Priority,
+		Queue:       insertRes.Queue,
+		ScheduledAt: insertRes.ScheduledAt,
+		State:       insertRes.State,
+		Tags:        insertRes.Tags,
+		UniqueKey:   insertRes.UniqueKey,
+	})
```

Comment:
This is a little unfortunate, but the addition of `(xmax != 0)` to `RETURNING *` means that sqlc won't return this as a `RiverJob` anymore, so we have to map it back to one as seen here. It's not great, but I copied over all the property tests from other insert functions so can verify we're not missing anything.

Reply:
Yeah, I always hate this effect when I want to select one or two little things in addition to the entire row. I sorta wish sqlc adopted an embedded struct for this situation instead of inventing all new structs with a few extra fields.

Reply:
Yeah, found myself wishing that _exactly_ thing. An option for an embedding struct when `*` was combined with something else would solve this problem completely.

Reply:
Hey @brandur, I think this is already possible when using `sqlc.embed`, check out this example: https://play.sqlc.dev/p/fa54ae82af605631d9ed837d0d71b0d661876e891217ff0c35903888f4d26df1

Reply:
Oh, amazing! Had no idea this feature existed.

Totally works. Opened #464. Thanks for pointing it out!